AppKit/CPView.j

```diff
@@ -3392,6 +3431,153 @@ setBoundsOrigin:
 
 @end
 
+@implementation CPView (TrackingAreaAdditions)
+
+- (void)addTrackingArea:(CPTrackingArea)trackingArea
+{
+    // Consistency check
+    if (!trackingArea || [_trackingAreas containsObject:trackingArea])
+        return;
+    
+    if ([trackingArea _isReferenced])
+        [CPException raise:CPInternalInconsistencyException reason:"Tracking area has already been added to another view."];
+
+    [_trackingAreas addObject:trackingArea];
+    [trackingArea _setReferencingView:self];
+    
+    if (_window)
+        [_window _addTrackingArea:trackingArea];
+  
+    [trackingArea _updateActualRect];
+}
+
+- (void)removeTrackingArea:(CPTrackingArea)trackingArea
+{
+    // Consistency check
+    if (!trackingArea)
+        return;
+    
+    if (![_trackingAreas containsObject:trackingArea])
+        [CPException raise:CPInternalInconsistencyException reason:"Trying to remove unreferenced trackingArea"];
+
+    [self _removeTrackingArea:trackingArea];
+}
+
+// Invoked automatically when the view’s geometry changes such that its tracking areas need to be recalculated.
+
+- (void)updateTrackingAreas
+{
+    // You should override this method to remove out of date tracking areas and add recomputed tracking areas;
+    
+    // Cocoa calls this on every view, whereas they have tracking area(s) or not.
+    // Cappuccino behaves differently :
+    // - updateTrackingAreas is called when placing a view in the view hierarchy (that is in a window)
+    // - if you have only CPTrackingInVisibleRect tracking areas attached to a view, it will not be called again (until you move the view in the hierarchy)
+    // - if you have at least one non-CPTrackingInVisibleRect tracking area attached, it will be called every time the view geometry could be modified
+    //   You don't have to touch to CPTrackingInVisibleRect tracking areas, they will be automatically updated
+    //
+    // Please note that it is the owner of a tracking area who is called for updateTrackingAreas.
+    // But, if a view without any tracking area is inserted in the view hierarchy (that is, in a window), the view is called for updateTrackingAreas.
+    // This enables you to use updateTrackingArea to initially attach your tracking areas to the view.
+}
+
+- (void)removeAllTrackingAreas
+{
+    // This utility method is intended for CPView subclasses overwriting updateTrackingAreas
+    //
+    // Typical use would be :
+    //
+    // - (void)updateTrackingAreas
+    // {
+    //      [self removeAllTrackingAreas];
+    //
+    //      ... add your specific updated tracking areas ...
+    //  }
+    
+    while (_trackingAreas.length > 0)
+        [self _removeTrackingArea:_trackingAreas[0]];
+}
+
+// Internal methods
+
+- (void)_removeTrackingArea:(CPTrackingArea)trackingArea
+{
+    if (_window)
+        [_window _removeTrackingArea:trackingArea];
+    
+    [trackingArea _setReferencingView:nil];
+    
+    [_trackingAreas removeObject:trackingArea];
```

Comment:
i'm sure. It would not be a bug to use removeObject: because _trackingArea cannot contains equal objects but it is faster and more explicit to use removeObjectIdenticalTo: